src/main/kotlin/blackjack/BlackJackGame.kt

```diff
@@ -14,30 +15,44 @@ class BlackJackGame {
         val players = Players(names)
         DisplayView.dealOutCards(players)
 
-        val dealer = Dealer(CardDeck())
-        dealer.dealOutCards(players)
-        DisplayView.cardsOfPlayers(players)
+        val distributor = Distributor(CardDeck())
+        val dealer = Dealer()
 
-        dealOutAdditionalCards(dealer, players)
-        DisplayView.result(players)
+        distributor.dealOutCards(dealer, players)
+        DisplayView.cardsOfPlayers(dealer, players)
+
+        dealOutAdditionalCards(distributor, players)
+        dealOutAdditionalCard(distributor, dealer)
+        DisplayView.finalScore(dealer, players)
+
+        GameResultCalculator.getResult(dealer, players)
+        DisplayView.result(dealer, players)
     }
 
-    private fun dealOutAdditionalCards(dealer: Dealer, players: Players) {
+    private fun dealOutAdditionalCards(distributor: Distributor, players: Players) {
         players.players.forEach {
-            dealOutAdditionalCard(dealer, it)
+            dealOutAdditionalCard(distributor, it)
         }
     }
 
-    private fun dealOutAdditionalCard(dealer: Dealer, player: Player) {
+    private fun dealOutAdditionalCard(distributor: Distributor, player: Player) {
         DisplayView.dealOutAdditionalCard(player)
         if (InputView.inputAdditionalCard() == "y") {
-            dealer.dealOutCard(player)
-            takeAnotherCard(dealer, player)
+            distributor.dealOutCard(player)
+            takeAnotherCard(distributor, player)
+        }
+    }
+
+    private fun dealOutAdditionalCard(distributor: Distributor, dealer: Dealer) {
+        val receiveNewCard = dealer.getScore() < Dealer.LIMIT_SCORE
+        if (receiveNewCard) {
+            distributor.dealOutCard(dealer)
         }
```

Comment:
점수를 가져온 뒤 BlackJackGame과 같은 입출력이 포함된 부분에서 비교 후 Boolean 값을 추출하기 보다는 해당 상태를 딜러 클래스에서 바로 가져올 수 있도록 구현하면 어떨까요?

```suggestion
        if (dealer.isReceiveNewCard) {
            distributor.dealOutCard(dealer)
        }
```